### PR TITLE
Functional Tests : API - Authorization Endpoint

### PR DESCRIPTION
**Pull request by Progi1984**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | 8.1.x
| Description?      | Functional Tests : API - Authorization Endpoint
| Type?             | improvement
| Category?         | TE
| BC breaks?        | no
| Deprecations?     | no
| How to test?      | CI is :green_circle: & Nightly is :green_circle: https://github.com/Progi1984/ga.tests.ui.pr/actions/runs/4788577051
| Fixed ticket?     | N/A
| Related PRs       | PrestaShop/keycloak_connector_demo#5
| Sponsor company   | @PrestaShopCorp

Thanks https://www.mastertheboss.com/keycloak/how-to-use-keycloak-admin-rest-api/

### CODE REVIEW

File: tests/UI/campaigns/functional/API/01_clientCredentialGrantFlow/02_externalAuthServer/01_authorizationEndpoint.ts [hunk 1]
```diff
@@ -0,0 +1,134 @@
+import api from '@utils/api';
+import keycloakHelper from '@utils/keycloakHelper';
+import testContext from '@utils/testContext';
+
+import {expect} from 'chai';
+import {APIRequestContext, request} from 'playwright';
+import {installKeycloak, createKeycloakClient, uninstallKeycloak} from '@commonTests/API/keycloak';
```
Comment (nesrineabdmouleh) [inline tests/UI/campaigns/functional/API/01_clientCredentialGrantFlow/02_externalAuthServer/01_authorizationEndpoint.ts]: => 'installKeycloak' is defined but never used  
Please fix the lint errors

File: tests/UI/utils/keycloakHelper.ts [hunk 2]
```diff
@@ -0,0 +1,88 @@
+import {APIRequestContext, APIResponse, Page} from 'playwright';
+
+/**
+ * @description Helper to use Keycloak
+ */
+export default {
+
+  async login(page: Page, username: string, password: string): Promise<void> {
```
Comment (nesrineabdmouleh) [inline tests/UI/utils/keycloakHelper.ts]: I think the JS doc is missing and maybe it's better to create variables for selectors wdyt ?

File: tests/UI/campaigns/functional/API/01_clientCredentialGrantFlow/02_externalAuthServer/01_authorizationEndpoint.ts (hunk 1 above)
Comment (Progi1984) [reply tests/UI/campaigns/functional/API/01_clientCredentialGrantFlow/02_externalAuthServer/01_authorizationEndpoint.ts]: Fixed

File: tests/UI/utils/keycloakHelper.ts (hunk 2 above)
Comment (Progi1984) [reply tests/UI/utils/keycloakHelper.ts]: Done

**Review by nesrineabdmouleh — DISMISSED** [review]

File: tests/UI/utils/keycloakHelper.ts [hunk 3]
```diff
@@ -0,0 +1,139 @@
+import {APIRequestContext, APIResponse, Page} from 'playwright';
```
Comment (boubkerbribri) [inline tests/UI/utils/keycloakHelper.ts]: I think this is 1/2 helper and 1/2 a page and we should separate them.
`login` => page function
`getClientSecret` => helper function

Another question : I'm not a fan of using and relying on another tool UI to setup something. Could we an API or some js api client for this ?

Comment (Progi1984) [reply tests/UI/utils/keycloakHelper.ts]: Fixed

**Review by nesrineabdmouleh — APPROVED** [review]

**Review by boubkerbribri — APPROVED** [review]
LGTM :tada: :tada: :heavy_check_mark:

**Review by nicosomb — APPROVED** [review]

**Pull request merged**